CRM/MembershipExtras/Job/OfflineAutoRenewal/SingleInstalmentPlan.php

```diff
@@ -46,9 +87,12 @@ protected function getRecurringContributions() {
          AND msl.auto_renew = 1
          AND msl.is_removed = 0
          AND msl.end_date IS NULL
-         AND ccr.next_sched_contribution_date IS NOT NULL
-         AND DATE(ccr.next_sched_contribution_date) <= DATE_ADD(CURDATE(), INTERVAL {$daysToRenewInAdvance} DAY)
     GROUP BY ccr.id
+      HAVING MIN(cm.end_date) <= DATE_ADD(CURDATE(), INTERVAL {$daysToRenewInAdvance} DAY)
+      OR (
+        COUNT(cm.id) = 0
+        AND next_run_date <= DATE_ADD(CURDATE(), INTERVAL {$daysToRenewInAdvance} DAY)
+      )
```

Comment:
With the `phpdoc` above made it clear to me what this query is doing, but I can see it became a bit complex and this query shares some parts with the `getRecurringContributions`'s query.